src/arrayMethodSort.js

```diff
@@ -4,8 +4,43 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    let compareFunctionFinished;
+
+    if (!compareFunction) {
+      compareFunctionFinished = (a, b) => {
+        const stringA = a.toString();
+        const stringB = b.toString();
+
+        if (stringA.match(/\p{Lu}/u) && !stringB.match(/\p{Lu}/u)) {
+          return -1;
+        } else if (stringB.match(/\p{Lu}/u) && !stringA.match(/\p{Lu}/u)) {
+          return 1;
+        } else {
+          return stringA.localeCompare(stringB, undefined, {
+            sensitivity: 'base',
+          });
+        }
```

Comment:
The default compare function is handling capital letters differently by moving them to the front. This is not the default behavior of Array.prototype.sort, which does not differentiate between uppercase and lowercase when sorting strings unless specified by the localeCompare options. To align with the default sort behavior, consider using localeCompare without the uppercase check.

Reply:
i removed options in localeCoampare but saved match comparison, because test 'sorts mixed case strings' needs this